xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/skin/ObjectPropertyWikiResource.java

```diff
@@ -59,10 +62,13 @@ public String getURL(XWikiDocument document) throws Exception
     {
         XWikiContext xcontext = this.xcontextProvider.get();
 
+        Map<String, String> parameters = new LinkedHashMap<>();
+        parameters.put(XWiki.CACHE_VERSION, document.getVersion());
```

Comment:
The version seemed obvious to me but I'm now wondering if the document date (or maybe both version and date) would not be safer. Thinking about use cases where the document is replaced with its history (from a backup for example) in which case you can end up with the same version associated to totally different content.

Reply:
If I follow you, you're basically saying that using the docVersion in the JSX/SSX is not necessarily enough either?
I'd prefer to have a consistent behaviour, and to modify it everywhere if we feel the need.

Reply:
>  I'd prefer to have a consistent behaviour, and to modify it everywhere if we feel the need.

I totally agree without.

Note that I'm not against the current version, I'm just wondering if we should also include (or replace with) the date or not.